xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/main/java/org/xwiki/refactoring/internal/listener/BackLinkUpdaterListener.java

```diff
@@ -79,41 +85,77 @@ public class BackLinkUpdaterListener extends AbstractEventListener
     @Inject
     private JobProgressManager progressManager;
 
+    @Inject
+    private JobContext jobContext;
+
     /**
      * Default constructor.
      */
     public BackLinkUpdaterListener()
     {
-        super(NAME, new DocumentRenamedEvent());
+        super(NAME, new DocumentRenamedEvent(), new DocumentDeletedEvent());
     }
 
     @Override
     public void onEvent(Event event, Object source, Object data)
     {
         if (event instanceof DocumentRenamedEvent) {
-            boolean updateLinks = true;
-            boolean updateLinksOnFarm = true;
-            Predicate<EntityReference> canEdit =
-                entityReference -> this.authorization.hasAccess(Right.EDIT, entityReference);
-
-            if (source instanceof MoveJob) {
-                MoveRequest request = (MoveRequest) data;
-                updateLinks = request.isUpdateLinks();
-                updateLinksOnFarm = request.isUpdateLinksOnFarm();
-                // Check access rights taking into account the move request.
-                canEdit = entityReference -> ((MoveJob) source).hasAccess(Right.EDIT, entityReference);
-            }
+            maybeUpdateLinksAfterRename(event, source, data);
+        } else if (event instanceof DocumentDeletedEvent && this.jobContext.getCurrentJob() instanceof DeleteJob) {
+            maybeUpdateLinksAfterDelete(event);
+        }
+    }
 
-            if (updateLinks) {
-                updateBackLinks((DocumentRenamedEvent) event, canEdit, updateLinksOnFarm);
-            }
+    private void maybeUpdateLinksAfterDelete(Event event)
+    {
+        DeleteJob job = (DeleteJob) this.jobContext.getCurrentJob();
+        DeleteRequest request = (DeleteRequest) job.getRequest();
+        Predicate<EntityReference> canEdit = entityReference -> (job.hasAccess(Right.EDIT, entityReference));
+        DocumentDeletedEvent deletedEvent = (DocumentDeletedEvent) event;
+
+        // In case the delete affected the child pages too, only the root document should have the links updated.
```

Comment:
Why? Again, there's no notion of a "root" document for a delete action. The delete job deletes the **list** of documents specified by the delete job request.